server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -147,26 +153,113 @@ private Setting<?> createSetting(
                     ? Setting.boolSetting(key, (boolean) defaultValue, propertyArray)
                     : Setting.boolSetting(key, (Setting<Boolean>) fallback.getSetting(), propertyArray);
             case Integer:
+                if (minValue != null && maxValue != null) {
```

Comment:
Since you haven't implemented the write portion, I don't know when or if you'd get null.  But I think for most of these numbers there are `MIN_VALUE` for a default min, and `MAX_VALUE` for a default max, that you will always have a value for it and not have to handle nulls.